src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((groupedMap, item) => {
+    const keyValue: any = item[key];
+    const keyString = String(keyValue);
```

Comment:
nice spoting here of type safety (that is, we don't need to add keyString: string because String function ensures that!)